src/TodoApp/Request.hs

```diff
@@ -52,95 +55,117 @@ data Request r where
   -- | Remove all tasks
   Reset :: Request ()
 
+data Connection
+  = TCP URI
+  | UnixSocket FilePath
+
+createUnixSocketManager :: FilePath -> IO HC.Manager
+createUnixSocketManager socketPath = HC.newManager $ HC.defaultManagerSettings
+  { HC.managerRawConnection = return $ \_ _ _ -> do
+      sock <- NS.socket NS.AF_UNIX NS.Stream NS.defaultProtocol
+      NS.connect sock (NS.SockAddrUnix socketPath)
+      HC.socketConnection sock 8192
+  }
+
 runRequest ::
   (MonadIO m, m ~ IO) =>
   Request a ->
-  URI ->
+  Connection ->
   m a
-runRequest = \case
-  Complete id -> complete id
-  ViewAll -> viewAll
-  View -> view
-  Add task -> add task
-  Delete id -> delete id
-  Reset -> reset
+runRequest req conn = case req of
```

Comment:
```suggestion
runRequest conn req = \case
```

Reply:
Likewise, flip arguments in the those functions. Generally, these args are put at front to enable currying.

Reply:
https://github.com/juspay/todo-app/pull/20/commits/ea8cf655a9c15ff15a1384949b329319b2ef14d2